sunpy/image/transform.py

```diff
@@ -325,6 +325,58 @@ def _rotation_skimage(image, matrix, shift, order, missing, clip):
     return rotated_image
 
 
+@add_rotation_function("cv2",
+                       handles_clipping=False, handles_image_nans=False, handles_nan_missing=False)
+def _rotation_skimage(image, matrix, shift, order, missing, clip):
+    """
+    * Rotates using :func:`cv2.warpAffine`
+    * The ``order`` parameter selects from the following interpolation algorithms:
+      * 0: nearest-neighbor interpolation
+      * 1: bilinear interpolation
+      * 2: bilinear interpolation, same as ``order=1``
+      * 3: bicubic interpolation
+      * 4: bicubic interpolation, same as ``order=3``
+      * 5: bicubic interpolation, same as ``order=3``
+    * An input image with integer data is cast to floats prior to passing to
+      :func:`~cv2.warpAffine`.  The output image can be re-cast using
+      :meth:`numpy.ndarray.astype` if desired.
+    """
+    try:
+        import cv2
+    except ImportError:
+        raise ImportError("The opencv-python package is required to use this rotation method.")
+
+    _CV_ORDER_FLAGS = {
+        0: cv2.INTER_NEAREST,
+        1: cv2.INTER_LINEAR,
+        2: cv2.INTER_LINEAR,
+        3: cv2.INTER_CUBIC,
+        4: cv2.INTER_CUBIC,
+        5: cv2.INTER_CUBIC,
+    }
+
+    if issubclass(image.dtype.type, numbers.Integral):
+        warn_user("Integer input data has been cast to float64.")
+        adjusted_image = image.astype(np.float64)
+    else:
+        adjusted_image = image.copy()
+
+    trans = np.empty((2, 3))
+    trans[:, :2] = matrix.T
+    trans[:, 2] = [shift[1], shift[0]]
+
+    # Swap the byte order if it is non-native (e.g., big-endian on a little-endian system)
+    if adjusted_image.dtype.byteorder == ('>' if sys.byteorder == 'little' else '<'):
+        adjusted_image = adjusted_image.byteswap().newbyteorder()
+
+    # missing must be a Python float, not a NumPy float
+    missing = float(missing)
+
+    return cv2.warpAffine(adjusted_image.T, trans, adjusted_image.shape,
+                          flags=_CV_ORDER_FLAGS[order] | cv2.WARP_INVERSE_MAP,
```

Comment:
What does `_CV_ORDER_FLAGS[order] | cv2.WARP_INVERSE_MAP` mean for the final input to the function?

Reply:
They're independent bits in the bitfield.  Bits 0-3 specify the interpolation algorithm, and bit 4 specifies whether the mapping is an inverse map.